scRNA-seq-advanced/01-reading_filtering_CellRanger.Rmd

```diff
@@ -0,0 +1,484 @@
+
+The first is the `rowData` table, which includes the contents of the `features.tsv.gz` file from earlier.
+Let's look at this table, extracting it from the SCE object with the `rowData()` function and using `head()` to view only the first 6 rows.
+
+```{r rowdata}
+head(rowData(raw_sce))
+```
+
+You can see that this table includes an `ID` for each gene, which is usually the Ensembl gene ID, as well as the corresponding gene symbol in the `Symbol` column. 
+Finally there is a column for `Type`, which in this case is always "Gene Expression".
+If there were another modality of data that had been assayed in this experiment, there might be other values in this column, such as "Antibody Capture" for CITE-seq experiments.
+
+The second table is the `colData` table, which now corresponds to the `barcodes.tsv.gz` file, containing one row per column of the `counts` assay.
+We can look at the this table using the `colData()` function (and `head()` again to prevent printing the whole table):
+
+```{r coldata}
+head(colData(raw_sce))
+```
+Here we see that there are two columns: 
+- one has the path of the file that we read in, listed as `Sample` (you may not see the whole path in this display); this should be identical in all rows from a single sample.
+- the other column is the `Barcode`, which is the sequence that was used to identify each potential droplet for sequencing (and a numeric tag, in this case).
+
+
+### Filtering empty droplets
+
+Most of the barcodes in any given 10X experiment will not be seen at all, so our first step can be to filter this raw data to only the cells where there is at least one transcript that was counted with that barcode.
+
+To do this, we will use the `colSums()` function to quickly add up all the counts that correspond to each possible barcode, then filter our `raw_sce` down to just those columns.
+We will need to extract the `counts` matrix from our SCE object, which we can do using the `counts()` function, conveniently enough.
+
+```{r remove zeros}
+# sum columns from counts matrix
+barcode_counts <- colSums(counts(raw_sce))
+
+# filter SCE object to only rows with counts > 0
+raw_sce <- raw_sce[, which(barcode_counts > 0)]
+```
+
+Now we can look at how our SCE object has changed:
+
+```{r zero-removed SCE}
+raw_sce
+```
+
+Of course, barcodes with zero counts are not the only ones that will not correspond to droplets without cells in them. 
+Even if a cell does not have a cell in it, there will often be spurious reads from RNA sequences that were present in the extracellular solution, whether from the original sample or cells that were disrupted during preparation for single-cell analysis. 
+
+We can identify these in part by low transcript counts, but we can also be a bit more clever, and look at the transcript counts _from_ the lowest-count droplets to create an expected distribution of transcript counts in droplets that don't contain cells.
```

Comment:
Agree on splitting it up, but I try to avoid `actually`, so some small changes.
```suggestion
We could identify these barcodes simply as those with low transcript counts.
Or, we can be a bit more clever!
We can look at the transcript counts _from_ the lowest-count droplets to create an expected distribution of transcripts in droplets that don't contain cells.
```